crates/rpc/rpc/src/eth/api/call.rs

```diff
@@ -269,25 +277,59 @@ where
                 }
             }
             // new midpoint
-            mid_gas_limit = (highest_gas_limit + lowest_gas_limit) / 2;
+            mid_gas_limit = ((highest_gas_limit as u128 + lowest_gas_limit as u128) / 2) as u64;
         }
 
         Ok(U256::from(highest_gas_limit))
     }
+
+    pub(crate) async fn create_access_list_at(
+        &self,
+        request: CallRequest,
+        at: Option<BlockId>,
+    ) -> EthResult<AccessList> {
+        let (mut cfg, block, at) = self.evm_env_at(at.unwrap_or_default()).await?;
+        let state = self.state_at_block_id(at)?.ok_or_else(|| EthApiError::UnknownBlockNumber)?;
+
+        // we want to disable this in eth_call, since this is common practice used by other node
+        // impls and providers <https://github.com/foundry-rs/foundry/issues/4388>
+        cfg.disable_block_gas_limit = true;
+
+        let mut env = build_call_evm_env(cfg, block, request)?;
+        let mut db = SubState::new(State::new(state));
+
+        let mut inspector = AccessListInspector::default();
```

Comment:
we need to use the `new` function instead with the from and to address 

see

https://github.com/foundry-rs/foundry/blob/28b2ae6d5ad3e802b813b679e26bc15b9b8d230b/anvil/src/eth/backend/mem/mod.rs#L981-L994

Reply:
Why do we need an inspector to check this? The accounts read are accessible in the database (the first layer is just a cache DB, so it is populated every time something is read and starts out empty), and the written accounts are accessible in the changeset given by the executor after execution. I think it's more performant to just read these values versus using an inspector, since an inspector hooks into every opcode step, every call etc. (even if it does nothing there is still some overhead)

Reply:
ref https://github.com/foundry-rs/foundry/pull/2839

Reply:
Done